File: ExampleCompose/app/src/main/java/com/example/examplecompose/data/bus/remote/dto/BusStopResponse.kt
package com.example.examplecompose.data.bus.remote.dto

import com.google.gson.annotations.SerializedName

data class BusStopResponseEntity(
    @SerializedName("response")
    val res : BusStopResponse? = null,
)

data class BusStopResponse(
    @SerializedName("header")
    val header : HeaderResponse? = null,
    @SerializedName("body")
    val body : BodyResponse? = null
)

data class HeaderResponse(
    @SerializedName("resultCode")
    val resultCode : String?,
    @SerializedName("resultMsg")
    val resultMessage : String?
)

data class BodyResponse(
    @SerializedName("items")
    val items: ItemResponse?,
    @SerializedName("numOfRows")
    val numberOfRows : Int?,
    @SerializedName("pageNo")
    val pageNumber : Int?,
    @SerializedName("totalCount")
    val totalCount : Int?,
);

data class ItemResponse(
    @SerializedName("item")
    val item: List<BusStopInfo>?,
);

data class BusStopInfo(
    @SerializedName("citycode")
    val cityCode: Int?,
    @SerializedName("gpslati")
    val gpsLatitude: Double?,
    @SerializedName("gpslong")
    val gpsLongitude: Double?,
    @SerializedName("nodeid")
    val nodeId: String?,
    @SerializedName("nodenm")
    val nodeName: String?,
    @SerializedName("nodeno")
    val nodeNumber : Int?,
);

File: ExampleFlutter/example_flutter/lib/data/bus/remote/dto/bus_stop_response.dart
import 'dart:ffi';

import 'package:json_annotation/json_annotation.dart';

part 'bus_stop_response.g.dart';

@JsonSerializable()
class BusStopResponseEntity {
  @JsonKey(name: "response")
  final BusStopResponse? res;

  BusStopResponseEntity(this.res);

  factory BusStopResponseEntity.fromJson(Map<String, dynamic> json) => _$BusStopResponseEntityFromJson(json);

  Map<String, dynamic> toJson() => _$BusStopResponseEntityToJson(this);
}

@JsonSerializable()
class BusStopResponse {
  @JsonKey(name: "header")
  final HeaderResponse? header;
  @JsonKey(name: "body")
  final BodyResponse? body;

  BusStopResponse(this.header, this.body);

  factory BusStopResponse.fromJson(Map<String, dynamic> json) => _$BusStopResponseFromJson(json);

  Map<String, dynamic> toJson() => _$BusStopResponseToJson(this);
}

@JsonSerializable()
class HeaderResponse {
  @JsonKey(name : "resultCode")
  final String? code;
  @JsonKey(name: "resultMsg")
  final String? message;

  HeaderResponse(this.code, this.message);

  factory HeaderResponse.fromJson(Map<String, dynamic> json) => _$HeaderResponseFromJson(json);

  Map<String, dynamic> toJson() => _$HeaderResponseToJson(this);
}

@JsonSerializable()
class BodyResponse {
  @JsonKey(name : "items")
  final ItemResponse? item;
  @JsonKey(name: "numOfRows")
  final int? numberOfRows;
  @JsonKey(name: "pageNo")
  final int? pageNumber;
  @JsonKey(name: "totalCount")
  final int? totalCount;

  BodyResponse(this.item, this.numberOfRows, this.pageNumber, this.totalCount);

  factory BodyResponse.fromJson(Map<String, dynamic> json) => _$BodyResponseFromJson(json);

  Map<String, dynamic> toJson() => _$BodyResponseToJson(this);
}

@JsonSerializable()
class ItemResponse {
  @JsonKey(name: "item")
  final List<BusStopInfo>? items;

  ItemResponse(this.items);

  factory ItemResponse.fromJson(Map<String, dynamic> json) => _$ItemResponseFromJson(json);

  Map<String, dynamic> toJson() => _$ItemResponseToJson(this);
}

@JsonSerializable()
class BusStopInfo {
  @JsonKey(name: "citycode")
  final int? cityCode;
  @JsonKey(name: "gpslati")
  final dynamic gpsLatitude;
  @JsonKey(name: "gpslong")
  final dynamic gpsLongitude;
  @JsonKey(name : "nodeid")
  final String? nodeId;
  @JsonKey(name : "nodenm")
  final String? nodeName;

  BusStopInfo(this.cityCode, this.gpsLatitude, this.gpsLongitude, this.nodeId, this.nodeName);

  factory BusStopInfo.fromJson(Map<String, dynamic> json) => _$BusStopInfoFromJson(json);

  Map<String, dynamic> toJson() => _$BusStopInfoToJson(this);
}